### PR TITLE
doc: add yaml design doc

### PR DESCRIPTION
**Pull request by gxthrj**

### CODE REVIEW

File: doc/yaml_struct.md [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+
+### Define traffic distribution rules
+
+```yaml
+kind: Rule
+name: xxx-rules
+hosts:
+- "domain.com"
+gateways:
+- baidu-gw
```
Comment (membphis) [inline doc/yaml_struct.md]: choose another name, `baidu` is not good

File: doc/yaml_struct.md [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+  - destination:
+       port: 28002
+       host: baidu-server
+       subset: v2
+  label:
+    app: baidu
+    version: v2
+
+```
+|  object/field   | describition |
```
Comment (membphis) [inline doc/yaml_struct.md:67]: need a blank line before this table

File: doc/yaml_struct.md [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+  ips：
+  - 127.0.0.1
+  - 127.0.0.2
+- name: v2
+  selector:
+    labels:
+      tag: v2
+
+```
+|  object/field   | describition |
```
Comment (membphis) [inline doc/yaml_struct.md:95]: ditto

File: doc/yaml_struct.md [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+|  ----  | ----  |
+|  destinations  | Target service |
+| destinations.host | Target service internal host |
+| destinations.subsets | Target service collection |
+
+### Plug-in extension
+The logic of the plugin itself is not defined here, only known plugins will be referenced here
```
Comment (membphis) [inline doc/yaml_struct.md:103]: we'd better add a blank line between the title and context.

File: doc/yaml_struct.md (hunk 1 above)
Comment (gxthrj) [reply doc/yaml_struct.md]: change `baidu` to `foo`

**Review by membphis — APPROVED** [review]

**Pull request merged**